src/receivers/HTTPReceiver.ts

```diff
@@ -143,6 +147,7 @@ export default class HTTPReceiver implements Receiver {
         return defaultLogger;
       })();
     this.endpoints = Array.isArray(endpoints) ? endpoints : [endpoints];
+    this.routes = prepareRoutes(customRoutes);
```

Comment:
@stevengill @seratch Requesting eyes for this decision. Swapped out `customRoutes` being stored in their original form for a new `ReceiverRoutes` object that holds the mapping of each `path`/`method`/`handler` combination.

Reply:
Just to clarify, you are essentially caching routes in `this.routes`

Reply:
Yes, exactly. I stopped short of refactoring it so that the other routes were also present (installation URL, for example), but that's part of the reason I chose `routes` instead of `customRoutes`: it could potentially contain _all_ routes associated with the receiver at some point in the future.